crates/libcontainer/src/process/init/context.rs

```diff
@@ -0,0 +1,52 @@
+use std::collections::HashMap;
+use std::path::Path;
+
+use oci_spec::runtime;
+
+use super::Result;
+use crate::container::Container;
+use crate::error::MissingSpecError;
+use crate::namespaces::Namespaces;
+use crate::process::args::ContainerArgs;
+use crate::syscall::Syscall;
+use crate::{notify_socket, utils};
+
+pub(crate) struct InitContext<'a> {
+    pub(crate) spec: &'a runtime::Spec,
+    pub(crate) linux: &'a runtime::Linux,
+    pub(crate) process: &'a runtime::Process,
+    pub(crate) rootfs: &'a Path,
+    pub(crate) envs: HashMap<String, String>,
+    pub(crate) ns: Namespaces,
+    pub(crate) syscall: Box<dyn Syscall>,
+    pub(crate) notify_listener: &'a notify_socket::NotifyListener,
+    pub(crate) hooks: Option<&'a runtime::Hooks>,
+    pub(crate) container: Option<&'a Container>,
+    pub(crate) rootfs_ro: bool,
+}
+
+impl<'a> InitContext<'a> {
+    pub fn builder(args: &'a ContainerArgs) -> Result<Self> {
```

Comment:
Nit: the logic here is more like a `From` which does a translation, not a builder.